ledger/catchpointwriter_test.go

```diff
@@ -713,3 +631,142 @@ func TestFullCatchpointWriterOverflowAccounts(t *testing.T) {
+	}
+
+	// At least 32 more blocks so that we catchpoint after the accounts exist
+	for i := 0; i < 40; i++ {
+		selfpay := pay
+		selfpay.Receiver = addrs[0]
+		selfpay.Note = ledgertesting.RandomNote()
+		dl.fullBlock(&selfpay)
+	}
+
+	tempDir := t.TempDir()
+
+	catchpointDataFilePath := filepath.Join(tempDir, t.Name()+".data")
+	catchpointFilePath := filepath.Join(tempDir, t.Name()+".catchpoint.tar.gz")
+
+	cph := testWriteCatchpoint(t, dl.validator.trackerDB().Rdb, catchpointDataFilePath, catchpointFilePath)
+	require.EqualValues(t, cph.TotalChunks, 1)
+
+	l := testNewLedgerFromCatchpoint(t, catchpointFilePath)
+	defer l.Close()
+}
+
+func TestCatchpointAfterTxns(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	genBalances, addrs, _ := ledgertesting.NewTestGenesis()
+	dl := NewDoubleLedger(t, genBalances, protocol.ConsensusFuture)
+	defer dl.Close()
+
+	boxApp := dl.fundedApp(addrs[1], 1_000_000, boxAppSource)
+	callBox := txntest.Txn{
+		Type:          "appl",
+		Sender:        addrs[2],
+		ApplicationID: boxApp,
+	}
+
+	makeBox := callBox.Args("create", "xxx")
+	makeBox.Boxes = []transactions.BoxRef{{Index: 0, Name: []byte("xxx")}}
+	dl.txn(makeBox)
+
+	pay := txntest.Txn{
+		Type:     "pay",
+		Sender:   addrs[0],
+		Receiver: addrs[1],
+		Amount:   100000,
+	}
+	// There are 12 accounts in the NewTestGenesis, plus 1 app account, so we
+	// create more so that we have exactly one chunk's worth, to make sure that
+	// works without an empty chunk between accounts and kvstore.
+	for i := 0; i < (BalancesPerCatchpointFileChunk - 13); i++ {
+		newacctpay := pay
+		newacctpay.Receiver = ledgertesting.RandomAddress()
+		dl.fullBlock(&newacctpay)
+	}
+	for i := 0; i < 40; i++ {
+		dl.fullBlock(pay.Noted(strconv.Itoa(i)))
+	}
+
+	tempDir := t.TempDir()
+
+	catchpointDataFilePath := filepath.Join(tempDir, t.Name()+".data")
+	catchpointFilePath := filepath.Join(tempDir, t.Name()+".catchpoint.tar.gz")
+
+	cph := testWriteCatchpoint(t, dl.validator.trackerDB().Rdb, catchpointDataFilePath, catchpointFilePath)
+	require.EqualValues(t, 2, cph.TotalChunks)
+
+	l := testNewLedgerFromCatchpoint(t, catchpointFilePath)
+	defer l.Close()
+	values, err := l.LookupKeysByPrefix(l.Latest(), "bx:", 10)
+	require.NoError(t, err)
+	require.Len(t, values, 1)
+
+	// Add one more account
+	newacctpay := pay
+	last := ledgertesting.RandomAddress()
+	newacctpay.Receiver = last
+	dl.fullBlock(&newacctpay)
+
+	// Write and read back in, and ensure even the last effect exists.
+	cph = testWriteCatchpoint(t, dl.validator.trackerDB().Rdb, catchpointDataFilePath, catchpointFilePath)
+	require.EqualValues(t, cph.TotalChunks, 2) // Still only 2 chunks, as last was in a recent block
+
+	/*
+		l = testNewLedgerFromCatchpoint(t, catchpointFilePath)
+		defer l.Close()
+		_, _, algos, err := l.LookupLatest(last)
+		// This doesn't work.  replay() probably isn't happening here. Figure out
+		// how to test
+		require.Equal(t, algos, 1_000_000)
+
+	*/
```

Comment:
@jannotti As per our verbal, I opened a child PR to enable the commented out assertion in https://github.com/algorand/go-algorand/pull/4596.  Can you review to see if you'd like to pull it in?